client/src/components/TODOList/TODOCalendarIcon.js

```diff
@@ -11,10 +11,26 @@ import TODOItem from "./TODOItem";
 // TODOItems starting at that date. Clicking the icon will show a popover
 // list of all the TODOItems for that "Date" tile
 const TODOCalendarIcon = ({ TODOList }) => {
+  // If TODOList contains at least one non-finished TODOItem,
+  // the icon applied to the current date tile will be red.
+  // If TODOList contains zero non-finished TODOItems, the icon
+  // applied to the current date tile will be grey.
+  const configureTODOIconColour = () => {
+    const notFinishedTODOs = TODOList.filter((todo) => {
+      return todo.isFinished === false;
+    });
+
+    if (notFinishedTODOs.length > 0) {
+      return "fill-red-500";
+    }
+
+    return "fill-gray-400";
```

Comment:
The new color assignment functionality seems to be working as intended so I would approve off of that. However, I did some inspect element toggling and seemingly managed to get the number disappearing due to styling fixed. Perhaps you can reflect the changes I made here in your css file:

![image](https://github.com/awctw/study-dash/assets/56493743/ccb67411-1f21-4e9d-a196-461061d4a235)

That transparency on hover makes the number disappear on hover, the transparency on the calendar tile makes the whole tile transparent, and setting the background-color and color to transparent when the tile is active makes the number disappear when the date is selected in general.

Reply:
wait... why is do I see blue in your image? I tried to get rid of that earlier

Reply:
I have fixed it all haha